test/tools/libtesteth/ImportTest.cpp

```diff
@@ -164,18 +163,49 @@ void ImportTest::makeBlockchainTestFromStateTest(set<eth::Network> const& _netwo
     }  // transactions
 }
 
-bytes ImportTest::executeTest()
+/// returns all networks that are defined in all expect sections
+set<eth::Network> ImportTest::getAllNetworksFromExpectSections(json_spirit::mArray const& _expects)
 {
-	assert(m_envInfo);
+    set<string> allNetworks;
+    for (auto const& exp : _expects)
+    {
+        requireJsonFields(exp.get_obj(), "expect",
+            {{"indexes", jsonVType::obj_type}, {"network", jsonVType::array_type},
+                {"result", jsonVType::obj_type}});
+        ImportTest::parseJsonStrValueIntoSet(exp.get_obj().at("network"), allNetworks);
+    }
+
+    allNetworks = test::translateNetworks(allNetworks);
+    set<eth::Network> networkSet;
+    for (auto const& net : allNetworks)
+        networkSet.emplace(test::stringToNetId(net));
+    return networkSet;
+}
 
+bytes ImportTest::executeTest(bool _isFilling)
+{
+    assert(m_envInfo);
     set<eth::Network> networks;
     if (!Options::get().singleTestNet.empty())
         networks.emplace(stringToNetId(Options::get().singleTestNet));
     else
-		networks = test::getNetworks();
+    {
+        if (_isFilling)
+        {
+            // Run tests only on networks from expect sections
+            BOOST_REQUIRE(m_testInputObject.count("expect") > 0);
+            networks = getAllNetworksFromExpectSections(m_testInputObject.at("expect").get_array());
+        }
+        else
+        {
+            // Run tests only on networks that are in post state of the filled test
+            for (auto const& post : m_testInputObject.at("post").get_obj())
+                networks.emplace(test::stringToNetId(post.first));
+        }
+    }
```

Comment:
Why not
```
else if (_isFilling)
{
   ...
}
else
{
  ...
}
```

Reply:
change after change after change. 
would you like to refactor this?

Reply:
That's how we work on code, no?